src/Microsoft.IdentityModel.Tokens/InMemoryCryptoProviderCache.cs

```diff
@@ -163,9 +221,23 @@ public override bool TryGetSignatureProvider(SecurityKey securityKey, string alg
 
             var cacheKey = GetCacheKeyPrivate(securityKey, algorithm, typeofProvider);
             if (willCreateSignatures)
+            {
+#if NET461 || NET472 || NETSTANDARD2_0 
                 return _signingSignatureProviders.TryGetValue(cacheKey, out signatureProvider);
+#elif NET45
+                _signingSignatureProviders.TryGetValue(cacheKey, out signatureProvider);
+                return signatureProvider != null;
+#endif
+            }
             else
+            {
+#if NET461 || NET472 || NETSTANDARD2_0 
                 return _verifyingSignatureProviders.TryGetValue(cacheKey, out signatureProvider);
+#elif NET45
+                _verifyingSignatureProviders.TryGetValue(cacheKey, out signatureProvider);
+                return signatureProvider != null;
```

Comment:
can this not be the same as directly above and return the TryGetValue's bool directly?

Reply:
Good catch! Fixed!